src/Middleware/StaticFiles/src/StaticFilesEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.StaticFiles;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    /// <summary>
+    /// Contains extension methods for using static files with endpoint routing.
+    /// </summary>
+    public static class StaticFilesEndpointRouteBuilderExtensions
+    {
+        /// <summary>
+        /// Registers a fallback <see cref="RouteEndpoint"/> that will serve the provided static file
+        /// </summary>
+        /// <param name="builder">The <see cref="IEndpointRouteBuilder"/>.</param>
+        /// <param name="filePath">The file path of the file to serve.</param>
+        /// <returns>The <see cref="IEndpointRouteBuilder"/></returns>
+        /// <remarks>
+        /// <para>
+        /// <see cref="MapFallbackToFile(IEndpointRouteBuilder, string)"/> is intended to handle cases where URL path of
+        /// the request does not contain a filename, and no other endpoint has matched. This is convenient for routing
```

Comment:
"does not contain a filename"

Should this be does?

Reply:
nope - it's correct.

This scenario is:
1. Requests for a file get handled by static file middleware
2. Requests for a non-file get re-routed to the SPA (which happens to a static file in this case)
3. Requests for a file that don't exist on disk result in a 404.